mutant/modules/artic_nanopore/parser.py

```diff
@@ -0,0 +1,211 @@
+        """Calculates fraction of N bases in a fasta file"""
+        with open(input_file, "r") as fasta:
+            total_bases = 0
+            total_N = 0
+            for line in fasta:
+                stripped_line = line.strip()
+                total_bases += len(stripped_line)
+                total_N += stripped_line.count('N')
+            percentage_N_two_decimals = round((total_N/total_bases)*100, 2)
+            return percentage_N_two_decimals
+
+    def parse_assembly(self, results: dict, resdir: str, barcode_to_sample: dict) -> dict:
+        """Collects data by parsing the assembly"""
+        base_path = "/".join([resdir, "articNcovNanopore_sequenceAnalysisMedaka_articMinIONMedaka"])
+        for filename in os.listdir(base_path):
+            if filename.endswith(".consensus.fasta"):
+                abs_path = os.path.join(base_path, filename)
+                first_line: str = self.get_line(filename=abs_path, line_index_of_interest=0)
+                cust_sample_id: str = self.get_cust_sample_id(line_to_parse=first_line, barcode_translation=barcode_to_sample)
+                fraction_N: float = self.get_fraction_n(input_file=abs_path)
+                results[cust_sample_id]["fraction_n_bases"] = fraction_N
+        return results
+
+    def get_depth_files_paths(self, barcode: str, base_path: str) -> list:
+        """Returns the paths to coverage statistics for a certain barcode"""
+        wild_card_path = "*".join([base_path, barcode, "depths"])
+        depth_files = glob.glob(wild_card_path)
+        return depth_files
+
+    def count_bases_w_10x_cov_or_more(self, coverage_stats: list) -> int:
+        """Counts how many times a number is higher than 10 in a list"""
+        numbers_higher_than_10 = 0
+        for number in coverage_stats:
+            if number >= 10:
+                numbers_higher_than_10 += 1
+        return numbers_higher_than_10
+
+    def initiate_coverage_stats_list(self, file_path: str) -> list:
+        """Collect coverage statistics"""
+        coverage_stats = []
+        with open(file_path, "r") as file1:
+            for line in file1:
+                stripped_line = line.strip()
+                columns: list = stripped_line.split("\t")
+                coverage_stats.append(int(columns[3]))
+        file1.close()
+        return coverage_stats
+
+    def calculate_coverage(self, results: dict, resdir: str, barcode_to_sample: dict) -> dict:
+        """Collects data for the fraction of each assembly that got 10x coverage or more"""
+        base_path = "/".join([resdir, "articNcovNanopore_sequenceAnalysisMedaka_articMinIONMedaka/"])
+        for barcode in barcode_to_sample:
+            coverage_files_paths: list = self.get_depth_files_paths(barcode=barcode, base_path=base_path)
+            coverage_stats: list = self.initiate_coverage_stats_list(file_path=coverage_files_paths[0])
+            with open(coverage_files_paths[1], "r") as file2:
```

Comment:
Just wondering here, why are there 2 depth files?

Reply:
I discussed this with Annick, but I don't really remember. I guess it is because it goes through the pore in two directions? What do you say @rannick ?

Reply:
I think it corresponds to the 2 primer sets, they are evaluated separately

Reply:
You can read about the 2 primer sets here, this also explains the mysterious 1s and 2s that have been confusing me in the primer.bed fil (5th column where score usually is found): https://artic.readthedocs.io/en/latest/primer-schemes/